### PR TITLE
Add ASP.NET Core + Angular 2 project template

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
As discussed, this PR adds an ASP.NET Core project that hosts an Angular 2 single-page app. The contents are based on the current Yeoman generator (`generator-aspnetcore-spa` version 0.2.9). There's also a script in here that can be run to update the project to match any newer version of the Yeoman generator that gets published in the future.

We have a load of enhancements planned for this template, including updating the Angular 2 RTM as soon as the angular-universal package is updated to support it, as well as various improvements to the developer experience in Visual Studio specificially. But this is a good start as-is, and already contains all the features from the Yeoman generator, including server-side prerendering and webpack middleware integration (with hot module replacement, etc.).

### CODE REVIEW

**Comment by madskristensen** [comment]
Awesome. Thanks Steve!!

**Comment by wis** [comment]
Awesome. will this eventually use the Angular Cli? will it be as simple as merging the two Webpack configs and fixing routing, naming and all the other conventions? 

in general How can you have a good edit-save-compile-hotload cycle with Angualr-hmr using the Angular Cli in an Asp.net Core WebAPI project and serving the static Angular 2 App? should you have a Angular 2 project in the wwwroot directory and serve it with the Angular Cli during development along side the asp.net core webapi but configure it during development to consume the api from the kestrel port, and in production not?

Last night after they announced Angular Final, in Q&A the Angular Team said:

> - Will the core team dedicate more resources to the Angular CLI?
>   - Yes, the team is definitely committed to invest more on the CLI and improve it. One of the things that has been done currently is to finally finish the webpack migration which is now available.
> - Do you guys have plans to make the CLI more configurable? [13:26-15:30](https://www.youtube.com/watch?v=xTIWBXkpvDc&start=806&end=938)
>   - There are a lot of requests for that currently and the team is working on it and wanna support in the future. The current direction is to build an add-on system.
>     Some other current ideas are around to build a kind of an “eject button” which allows you to get out of the CLI at some point if you need way more flexibility.

File: Templates/AspNetCoreAngular2/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Welcome to ASP.NET 5
```
Comment (Eilon) [inline Templates/AspNetCoreAngular2/README.md:1]: Not ASP.NET 5 😄

File: Templates/AspNetCoreAngular2/Controllers/SampleDataController.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            public string DateFormatted { get; set; }
+            public int TemperatureC { get; set; }
+            public string Summary { get; set; }
+
+            public int TemperatureF
+            {
+                get
+                {
+                    return 32 + (int)(this.TemperatureC / 0.5556);
```
Comment (Eilon) [inline Templates/AspNetCoreAngular2/Controllers/SampleDataController.cs:39]: `this.NOT`

**Comment by SteveSandersonMS** [comment]
> Awesome. will this eventually use the Angular Cli?

Some parts of angular CLI are applicable when your app is hosted in ASP.NET Core, such as the `ng generate` commands, which we'll definitely make usable. Other parts of angular CLI would not be useful, such as `ng serve`, because ASP.NET is already a web server and does a slick job of integrating the server and client parts already (e.g., for server-side prerendering and webpack HMR integration).

**Comment by wis** [comment]
@SteveSandersonMS Cool, the progress is great, now we are up to date with the Angular 2.0.0 final release. 

what about bundling AOT app, and tree shaking and optimizing for production? 
check out [this blog post: "Building an Angular 2 Application for Production"](http://blog.mgechev.com/2016/06/26/tree-shaking-angular2-production-build-rollup-javascript/) which talks about the long optimizing process to get a maximally small, efficient production app ready to serve, this process is going to happen automatically with the angular-cli. 
it is being worked on by many contributors and much resources are being put into it. thus hypothetically speaking, the cli implementation will always be superior.

why can't the angular-cli be a dependency and delegate the building & bundling tasks to it?

the dev middleware is great as it is, the edit-auto-build-hot-reload cycle is awesome, can the cli at least be used for production build & bundle?

**Comment by SteveSandersonMS** [comment]
@wesamco For general suggestions about the Angular 2 template, or for reporting any issues with it, could I ask you to post to https://github.com/aspnet/JavaScriptServices/issues? That's where we're actually developing this template (and others), so you'll get more of a response there :)

Short answer to your question is that we'll use whatever's best. Once `angular-cli` reaches the point where it can do the things you've mentioned (AoT, tree-shaking, etc.) we'll definitely integrate the relevant parts, potentially delegating to that tool during builds or whatever APIs they offer. We're not tied to any specific technology pattern - we'll just do what works best. But it's too early for `angular-cli` right now (doesn't yet do the things you're talking about).

**Pull request merged**